policy/modules/services/incus.te

```diff
@@ -0,0 +1,198 @@
+
+########################################
+#
+# incus daemon local policy
+
+allow incusd_t self:process getattr;
+
+# incusd wants to load netdev-phys*
+kernel_request_load_module(incusd_t)
+
+ifdef(`init_systemd',`
+	init_start_system(incusd_t)
+	init_stop_system(incusd_t)
+	init_get_system_status(incusd_t)
+	init_stop_generic_units(incusd_t)
+')
+
+# comm="incusd" path="/run/systemd/resolve/stub-resolv.conf"
+# allow incusd_t container_tmpfs_t:file { open read };
+# allow incusd_t container_tmpfs_t:dir search;
+container_read_tmpfs_files(incusd_t)
+
+fs_getattr_binfmt_misc_fs(incusd_t)
+fs_getattr_binfmt_misc_dirs(incusd_t)
+fs_dontaudit_manage_fusefs_files(incusd_t)
+
+# here dontaudit would also work, but consuming more resources
+fs_watch_tmpfs_dirs(incusd_t)
+
+dnsmasq_domtrans(incusd_t)
+dnsmasq_signal(incusd_t)
+dnsmasq_kill(incusd_t)
+dnsmasq_signull(incusd_t)
+
+kernel_setsched(incusd_t)
+dpkg_dontaudit_manage_db(incusd_t)
+
+files_exec_usr_files(incusd_t)
+
+# no error, but this AVC:
+#  scontext="system_u:system_r:incusd_t:s0" #  tcontext="system_u:system_r:iptables_t:s0" #  class="process"
+#  perms="sigkill" #  comm="incusd" exe="" path=""
+# candidate for dontaudit instead?
+iptables_admin(incusd_t,system_r)
+
+# Error: Failed to run: /usr/bin/incusd forkstart first /var/lib/incus/containers /run/incus/first/lxc.conf: exit status 1
+mount_exec(incusd_t)
+
+# Error: Daemon still not running after 10s timeout...
+rsync_exec(incusd_t)
+
+# Error: Daemon still not running after 10s timeout...
+allow incusd_t self:anon_inode { create map read write };
+allow incusd_t self:io_uring sqpoll;
+allow incusd_t self:netlink_kobject_uevent_socket { read create setopt bind getattr write };
+allow incusd_t self:vsock_socket { connect read write accept bind create getattr getopt listen };
+
+# Error: Failed pre-start check for device "eth0": Network "incusbr0" unavailable on this server
+allow incusd_t self:rawip_socket { read write ioctl create getopt setopt };
+
+# Error: Failed instance creation: Failed to start device "eth0":
+#  Failed to create the tap interfaces "tape73a8425":
+#  Failed to run: ip tuntap add name tape73a8425 mode tap multi_queue:
+#  exit status 1 (ioctl(TUNSETIFF): Permission denied)
+allow incusd_t self:tun_socket { create relabelfrom relabelto };
+
+# avc:  denied  { read } for  pid=20404 comm="qemu-system-x86"
+# name="overcommit_memory" dev="proc" ino=16557
+# scontext=system_u:system_r:incusd_t:s0
+# tcontext=system_u:object_r:sysctl_vm_overcommit_t:s0 tclass=file permissive=0
+kernel_read_vm_overcommit_sysctl(incusd_t)
+
+# for bridge management
+dev_create_sysfs_files(incusd_t)
+dev_write_sysfs(incusd_t)
+
+# incusd wont start without access to /dev/urandom
+dev_read_urand(incusd_t)
+
+# mounton access to /proc/sys/kernel/random/boot_id
+kernel_mounton_kernel_sysctl_files(incusd_t)
+
+# read /etc/machine-id
+files_read_etc_runtime_files(incusd_t)
+
+# watch /dev/hugepages
+files_watch_all_dirs(incusd_t)
```

Comment:
Is this just for hugepages? We can create a new interface for it like `fs_watch_hugetlbfs_dirs()`.

Reply:
I removed the line noe because I cannot reproduce this denial anymore. Maybe due to an incus update, not sure. But if it reappears I will add that interface for it.